configure.ac

```diff
@@ -530,7 +530,10 @@ AC_ARG_WITH([qt5],
 AC_ARG_VAR([MOC], [full path to Qt moc program])
 
 PKG_CHECK_MODULES([QT],
-		  [Qt5Core >= 5 Qt5Gui >= 5 Qt5Widgets >= 5.0 Qt5X11Extras >= 5.0])
+		  [Qt5Core >= 5 Qt5Gui >= 5 Qt5Widgets >= 5.0])
+AS_IF([test "x$have_x" = "xyes"], [
+  PKG_CHECK_MODULES([QT], [Qt5X11Extras >= 5.0])
+])
 
```

Comment:
This caused a breakage on Linux. PKG_CHECK_MODULES() has an weird behavior. I did a local test: folding with this change makes it build locally again:

```
diff --git a/configure.ac b/configure.ac
index 9b34ec11f802..e73792c23274 100644
--- a/configure.ac
+++ b/configure.ac
@@ -540,11 +540,12 @@ AC_ARG_WITH([qt5],
 
 AC_ARG_VAR([MOC], [full path to Qt moc program])
 
+AS_IF([test "x$have_x" = "xyes"],
+      [qt_extra="Qt5X11Extras >= 5.0"],
+      [qt_extra=""])
+
 PKG_CHECK_MODULES([QT],
-		  [Qt5Core >= 5 Qt5Gui >= 5 Qt5Widgets >= 5.0])
-AS_IF([test "x$have_x" = "xyes"], [
-  PKG_CHECK_MODULES([QT], [Qt5X11Extras >= 5.0])
-])
+		  [Qt5Core >= 5 Qt5Gui >= 5 Qt5Widgets >= 5.0 $qt_extra])
 
 AS_IF([test "x$with_qt" != "xno"],
    AS_IF([test "x$with_qt5" != "xno"],
```

I folded it on my test branch and waiting for TravisCI results. if it passes, I'll merge it.

Reply:
FYI, at the tescription of PKG_CHECK_MODULES(), it is said that the prefix should be unique: https://autotools.io/pkgconfig/pkg_check_modules.html

Currently, there are some violations on such rule at configure.ac. One of the changes I did on the yesterday's patch is meant to address a problem caused by a bad usage of this macro.

Reply:
That sounds like a much better solution than what I added, thanks!